### PR TITLE
Connect to Cassandra cluster using SSL

### PR DESCRIPTION
**Pull request by SandishKumarHN**
Connect to Cassandra cluster with tls security

```
== RELEASE NOTES ==

Cassandra Changes
* Add TLS security support.
```

### CODE REVIEW

**Comment by dborkar** [comment]
@SandishKumarHN thank you for your contribution! This is something community has brought up. Can you share your testing approach? 
@highker Can you please take a look?

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        }
+        try {
+            // load KeyStore if configured and get KeyManagers
+            KeyStore keystore = null;
+            KeyManager[] keyManagers = null;
+            if (keystorePath.isPresent()) {
+                char[] keyManagerPassword;
+                try {
+                    // attempt to read the key store as a PEM file
+                    keystore = PemReader.loadKeyStore(keystorePath.get(), keystorePath.get(), keystorePassword);
```
Comment (highker) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java:76]: static import loadKeyStore

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        }
+    }
+
+    public KeyStore loadTrustStore(File trustStorePath, Optional<String> trustStorePassword)
+            throws IOException, GeneralSecurityException
+    {
+        KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        try {
+            // attempt to read the trust store as a PEM file
+            List<X509Certificate> certificateChain = PemReader.readCertificateChain(trustStorePath);
```
Comment (highker) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java:128]: same

File: presto-cassandra/pom.xml [hunk 3]
```diff
@@ -166,6 +166,10 @@
             <artifactId>testing</artifactId>
             <scope>test</scope>
         </dependency>
+      <dependency>
```
Comment (highker) [inline presto-cassandra/pom.xml:169]: break a line between these two. Also, the indentation doesn't look right.

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraErrorCode.java [hunk 4]
```diff
@@ -22,7 +22,8 @@
 public enum CassandraErrorCode
         implements ErrorCodeSupplier
 {
-    CASSANDRA_METADATA_ERROR(0, EXTERNAL), CASSANDRA_VERSION_ERROR(1, EXTERNAL);
+    CASSANDRA_METADATA_ERROR(0, EXTERNAL), CASSANDRA_VERSION_ERROR(1, EXTERNAL),
+    CASSANDRA_SSL_INITIALIZATION_FAILURE(2, EXTERNAL);
```
Comment (highker) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraErrorCode.java:26]: one error code per line

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java (hunk 1 above)
Comment (SandishKumarHN) [reply presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java:76]: @highker I did not understand this comment. PemReader has all static methods.

**Comment by SandishKumarHN** [comment]
@highker thanks for the review, I have made changes based on your review and left a question. looks like the CI failure is not related to the PR.

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java (hunk 2 above)
Comment (highker) [reply presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java:128]: import static com.facebook.airlift.security.pem.PemReader.readCertificateChain;

**Comment by SandishKumarHN** [comment]
@highker thanks for the review again, made changes for static import, and squshed into one commit. CI failure is not related to this PR.

**Pull request merged**